scalding-commons/src/main/scala/com/twitter/scalding/commons/source/LzoGenericScheme.scala

```diff
@@ -0,0 +1,128 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.twitter.scalding.commons.source
+
+import scala.reflect.ClassTag
+
+import com.twitter.bijection._
+import com.twitter.chill.Externalizer
+import com.twitter.elephantbird.cascading2.scheme.LzoBinaryScheme
+import com.twitter.elephantbird.mapreduce.input.combine.DelegateCombineFileInputFormat
+import com.twitter.elephantbird.mapreduce.io.{ BinaryConverter, GenericWritable }
+import com.twitter.elephantbird.mapreduce.input.{ BinaryConverterProvider, MultiInputFormat }
+import com.twitter.elephantbird.mapreduce.output.LzoGenericBlockOutputFormat
+import com.twitter.elephantbird.mapred.output.DeprecatedOutputFormatWrapper
+
+import org.apache.hadoop.mapred.{ JobConf, OutputCollector, RecordReader }
+import org.apache.hadoop.conf.Configuration
+
+import cascading.tap.Tap
+import cascading.flow.FlowProcess
+
+/**
+ * Serializes BinaryConverters to JobConf.
+ */
+private[source] object ExternalizerSerializer {
+  def inj[T]: Injection[Externalizer[T], String] = {
+    import com.twitter.bijection.Inversion.attemptWhen
+    import com.twitter.bijection.codec.Base64
+
+    implicit val baseInj = JavaSerializationInjection[Externalizer[T]]
+
+    implicit val unwrap: Injection[GZippedBase64String, String] =
+      // this does not catch cases where it's Base64 but not compressed
+      // but the decompression injection will, so it's safe to do this
+      new AbstractInjection[GZippedBase64String, String] {
+        override def apply(gzbs: GZippedBase64String) = gzbs.str
+        override def invert(str: String) = attemptWhen(str)(Base64.isBase64)(GZippedBase64String(_))
+      }
+
+    Injection.connect[Externalizer[T], Array[Byte], GZippedBase64String, String]
+  }
+}
+
+private[source] object ConfigBinaryConverterProvider {
+  val ProviderConfKey = "com.twitter.scalding.lzo.converter.provider"
+}
+
+/**
+ * Provides BinaryConverter serialized in JobConf.
+ */
+private[source] class ConfigBinaryConverterProvider[M] extends BinaryConverterProvider[M] {
+  import ConfigBinaryConverterProvider._
+
+  private[this] var cached: Option[(String, BinaryConverter[M])] = None
+
+  override def getConverter(conf: Configuration): BinaryConverter[M] = {
+    val data = conf.get(ProviderConfKey)
+    require(data != null, s"$ProviderConfKey is not set in configuration")
+    cached match {
+      case Some((data, conv)) => conv
+      case _ =>
+        val extern = ExternalizerSerializer.inj.invert(data).get
+        val conv = extern.get.asInstanceOf[BinaryConverter[M]]
+        cached = Some((data, conv))
+        conv
+    }
+  }
+}
+
+/**
+ * Generic scheme for data stored as lzo-compressed protobuf messages.
+ * Serialization is performed using the supplied BinaryConverter.
+ */
+class LzoGenericScheme[M: ClassTag](@transient conv: BinaryConverter[M]) extends LzoBinaryScheme[M, GenericWritable[M]] {
```

Comment:
is the ClassTag serializable? Have we tested that this actually round-trips a java serialization (which should happen when cascading uses this on Hadoop).

I wonder if instead of keeping the ClassTag here if it is safer to just take a Class, which is serializable, and make an apply method that uses ClassTag if needed.